llama-index-core/llama_index/core/base/llms/base.py

```diff
@@ -253,3 +254,19 @@ async def astream_complete(
                 print(response.text, end="", flush=True)
             ```
         """
+
+    def __init_subclass__(cls, **kwargs) -> None:
```

Comment:
Ohhhh this is nice!

Up to now, I've been resorting to decorating the subclasses individually. If I'm reading this correctly, then this new hook allows for "decorating" these methods with span automatically.

Reply:
Yes, all subclasses will have these methods decorated automatically.

Reply:
Lets add a docstring to this method to explain what's happening in here for future devs

Reply:
added